### ISSUE TITLE
SearchListControl has wrong parent-child roles relationship

### ISSUE THREAD

**Issue by Aljullu** [woocommerce/woocommerce-admin#3642]
**To Reproduce**
1. Install the Axe extension ([Firefox](https://addons.mozilla.org/en-US/firefox/addon/axe-devtools/), [Chrome](https://chrome.google.com/webstore/detail/axe-web-accessibility-tes/lhdoppojpmngadmnindnejefpokejbdd)) so you can do quick accessibility analyses.
2. Go to _WooCommerce_ > _DevDocs_ > scroll down to _SearchListControl_ > press _Toggle loading state_.
3. Go to the _Axe_ tab of your browser devtools and click on _Analyze_. You will see an error appear:
	>  Certain ARIA roles must be contained by particular parents

That's because [`<SearchListItem>` sets `role="menuitemradio"` or `role="menuitemcheckbox"`](https://github.com/woocommerce/woocommerce-admin/blob/master/packages/components/src/search-list-control/item.js#L65). Those two roles require the parent element to have `role="menu"` ([see example](https://www.w3.org/WAI/GL/wiki/Using_ARIA_menus#Success_Criteria_4.1.2_Name.2C_Role.2C_Value)) but the parent component is [`<MenuGroup>`](https://github.com/woocommerce/woocommerce-admin/blob/master/packages/components/src/search-list-control/index.js#L7) which has a hardcoded [`role="group"`](https://github.com/WordPress/gutenberg/blob/master/packages/components/src/menu-group/index.js#L40).

I think the way to proceed would be:
* Create a PR to `gutenberg` to add a `role` prop to `<MenuGroup>`.
* Update our current `<SearchListControl>` component to pass down a `role="menu"` to `MenuGroup`.

**Comment by jeffstieler** [comment]
What about adding the `role="menu"` here? https://github.com/WordPress/gutenberg/blob/master/packages/components/src/menu-group/index.js#L23

**Comment by Aljullu** [comment]
> What about adding the role="menu" here? https://github.com/WordPress/gutenberg/blob/master/packages/components/src/menu-group/index.js#L23

I did a quick test with devtools and it looks like that would work as well @jeffstieler.

**Comment by diegohaz** [comment]
Hello! I don't think replacing the accessibility semantics of `MenuGroup` is a good solution. More context on https://github.com/WordPress/gutenberg/pull/28053#pullrequestreview-564670716:

> While I agree that this component should accept not only the `role` prop, but also the other HTML attributes, I don't think replacing the accessibility semantics of this component is the best solution, especially because `role="menu"` elements require [specific behaviors that should be implemented by the developer](https://www.w3.org/TR/wai-aria-practices/#menu), and `MenuGroup` doesn't implement them.
>
> [`DropdownMenu`](https://developer.wordpress.org/block-editor/components/dropdown-menu/) does implement those behaviors. So I would say the best solution is to wrap `MenuGroup` with `DropdownMenu` (you can still do `DropdownMenu` -> `MenuGroup` -> `MenuItem` as it's a valid accessible structure).

**Comment by Aljullu** [comment]
Yeah... good point @diegohaz. Maybe the best solution would be to refactor `SearchListItem` completely? They shouldn't be buttons with the roles `menuitemcheckbox` and  `menuitemradio`, instead, they could be a list of items with real checkboxes or radio controls. That will require some markup and styling updates, but seems to be better a11y-wise.

**Comment by diegohaz** [comment]
Yeah! If there's no need for a `role="menu"` with the behaviors described in the WAI-ARIA docs, it's better to go for native checkbox and radio buttons.

There's also a specific section for this in the APG: [No ARIA is better than Bad ARIA](https://www.w3.org/TR/wai-aria-practices-1.1/#no_aria_better_bad_aria) 😄
